### ISSUE TITLE
aws_route cannot create IPv6 route via ENI

### ISSUE THREAD

**Issue by stevenewey** [terraform-providers/terraform-provider-aws#2264]
### Terraform Version
terraform 0.10.8 / terraform-provider-aws 0.2.0

### Affected Resource(s)
- aws_route

### Terraform Configuration Files
```hcl
provider "aws" {
  region = "us-east-1"
  version = "~> 1.2"
}

resource "aws_vpc" "examplevpc" {
  cidr_block = "10.100.0.0/16"
  enable_dns_hostnames = true
  assign_generated_ipv6_cidr_block = true
}

resource "aws_internet_gateway" "internet" {
  vpc_id = "${aws_vpc.examplevpc.id}"
}

resource "aws_route" "igw" {
  route_table_id = "${aws_vpc.examplevpc.main_route_table_id}"
  destination_cidr_block = "0.0.0.0/0"
  gateway_id = "${aws_internet_gateway.internet.id}"
}

resource "aws_route" "igw-ipv6" {
  route_table_id = "${aws_vpc.examplevpc.main_route_table_id}"
  destination_ipv6_cidr_block = "::/0"
  gateway_id = "${aws_internet_gateway.internet.id}"
}

resource "aws_subnet" "router-network" {
  cidr_block = "10.100.1.0/24"
  vpc_id = "${aws_vpc.examplevpc.id}"
  ipv6_cidr_block = "${cidrsubnet(aws_vpc.examplevpc.ipv6_cidr_block, 8, 1)}"
  assign_ipv6_address_on_creation = true
  map_public_ip_on_launch = true
  availability_zone = "us-east-1a"
}

resource "aws_subnet" "client-network" {
  cidr_block = "10.100.10.0/24"
  vpc_id = "${aws_vpc.examplevpc.id}"
  ipv6_cidr_block = "${cidrsubnet(aws_vpc.examplevpc.ipv6_cidr_block, 8, 2)}"
  assign_ipv6_address_on_creation = true
  map_public_ip_on_launch = false
  availability_zone = "us-east-1a"
}

resource "aws_route_table" "client-routes" {
  vpc_id = "${aws_vpc.examplevpc.id}"
}

resource "aws_route_table_association" "client-routes" {
  route_table_id = "${aws_route_table.client-routes.id}"
  subnet_id = "${aws_subnet.client-network.id}"
}

data "aws_ami" "ubuntu" {
  most_recent = true
  filter {
      name   = "name"
      values = ["ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-*"]
  }
  filter {
      name   = "virtualization-type"
      values = ["hvm"]
  }
  owners = ["099720109477"]
}

resource "aws_instance" "test-router" {
  ami = "${data.aws_ami.ubuntu.image_id}"
  instance_type = "t2.small"
  subnet_id = "${aws_subnet.router-network.id}"
}

resource "aws_network_interface" "router-internal" {
  subnet_id = "${aws_subnet.client-network.id}"
  source_dest_check = false
}

resource "aws_network_interface_attachment" "router-internal" {
  device_index = 1
  instance_id = "${aws_instance.test-router.id}"
  network_interface_id = "${aws_network_interface.router-internal.id}"
}

resource "aws_route" "internal-default-route" {
  route_table_id = "${aws_route_table.client-routes.id}"
  destination_cidr_block = "0.0.0.0/0"
  network_interface_id = "${aws_network_interface.router-internal.id}"
}

resource "aws_route" "internal-default-route-ipv6" {
  route_table_id = "${aws_route_table.client-routes.id}"
  destination_ipv6_cidr_block = "::/0"
  network_interface_id = "${aws_network_interface.router-internal.id}"
}
```

### Debug Output
https://gist.github.com/stevenewey/2765404731b1660e22f5cb312352f7b6

### Expected Behavior
A route table entry should be created for the IPv6 CIDR block via the specified ENI.

### Actual Behavior
An error occurs because the destinationIpv6CidrBlock is not included in the request made to the EC2 API:
```
Action=CreateRoute&DestinationCidrBlock=&NetworkInterfaceId=eni-XXX&RouteTableId=rtb-XXX&Version=2016-11-15
```
 
### Steps to Reproduce
The example above is self contained and can be used to produce the issue against an AWS account where the user already has credentials loaded.
1. terraform apply

**Comment by bflad** [comment]
Support for adding IPv6 routes to instances and network interfaces has been merged into master and will be released in v1.8.0 of the AWS provider (ETA end of week).

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.8.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#180-january-29-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
